text/0000-standardize-error-docs.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: standardize_error_docs
+- Start Date: 2023-01-10
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+
+# Summary
+[summary]: #summary
+
+(Re)-standardize long-form error code explanations (referred to in this RFC as "error docs") with a more flexible system that also takes into account new errors which do not fit into the format as defined by [RFC 1567]. This RFC replaces [RFC 1567] as the primary specification for error docs. This RFC ultimately wishes to make error docs more accessible and easy to read for both new and proficient Rust users.
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 1567] provided a standard that fulfilled its aim of ensuring *readable* and *helpful* error docs. However many changes have occurred to error docs since the RFC, such as [error docs being moved to their own files](https://github.com/rust-lang/rust/pull/66314), [error docs' examples use multiple crates](https://github.com/rust-lang/rust/pull/106028) and [documentation of internal error codes](https://github.com/rust-lang/rust/pull/106614). These changes, among others, now make [RFC 1567] out of date and re-creates the very problem [RFC 1567] solved: nonstandardized error docs.  
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC makes three changes to error docs in order to fulfill this RFC's (and [RFC 1567]'s) goals:
+
+- Firstly, that the style of error docs is not defined by RFC but by a "style spec" located in the `rust-lang/rust` repository. This style spec can be updated from time-to-time in order to ensure consistent formatting across error docs. Any change to the style spec must be with the consensus of the dev-tools team. It is recommended that the style spec is initially extracted from [RFC 1567].
```

Comment:
Might be worth giving the path to the style spec.

Reply:
Hmm, should we put it in `compiler/rustc_error_codes/STYLE_SPEC.md`?

Reply:
Just realized that it doesn't exist yet. Question for other reviewers: should this style spec be written before this RFC is merged or after is fine?

Reply:
> Hmm, should we put it in `compiler/rustc_error_codes/STYLE_SPEC.md`?

That seems like a good idea.